### PR TITLE
Add @gatsbyjs/reach-router types

### PR DESCRIPTION
**Pull request by domdomegg**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## gatsbyjs__reach-router ([<kbd>unpkg</kbd>](https://unpkg.com/browse/gatsbyjs__reach-router@latest/))
was missing the following properties:
1. matchPath
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against c7632fe994c8470725be9f9b72c7317c7610330c
</p>

**Review by Malena205 — APPROVED** [review]

File: types/gatsbyjs__reach-router/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+// Type definitions for @gatsbyjs/reach-router 1.3
+// Project: https://github.com/gatsbyjs/reach-router#readme
+// Definitions by: Adam Jones <https://github.com/domdomegg>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export * from 'reach__router';
```
Comment (peterblazejewicz) [inline types/gatsbyjs__reach-router/index.d.ts:6]:
```suggestion
export * from '@reach/router';
```

File: types/gatsbyjs__reach-router/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+            "../"
+        ],
+        "types": [],
+        "noEmit": true,
+        "forceConsistentCasingInFileNames": true,
+        "jsx": "react",
+        "paths": {
+            "@gatsbyjs/*": [
+                "gatsbyjs__*"
+            ]
```
Comment (peterblazejewicz) [inline types/gatsbyjs__reach-router/tsconfig.json:23]:
```suggestion
            ],
            "@reach/*": [
                "reach__*"
            ]
```

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@domdomegg thx!

**Comment by domdomegg** [comment]
Ready to merge

**Pull request merged**